Training/interface/DataLoader_main.h

```diff
@@ -194,6 +194,7 @@ class DataLoader {
             ("w_1_"+tau_name).c_str(),
             ("w_1_"+tau_name).c_str()
         ));
+        hist_weights[tau_type]->SaveAs(("Temp_"+tau_name+".root").c_str()); // It's required that all bins are filled in these histograms; save them to check incase binning is too fine and some bins are empty
```

Comment:
the histogram should be saved only when debug mode is on (e.g. add `debug` parameter in .yaml config)

Reply:
Ok, I fixed this (didn't push yet), but I'm wondering: Is there a way to check if a `debug` parameter is initialized in the yaml config? Right now I have an additional line to check for this in Trainingv0p1.py and apply_training.py (would it be needed anywhere else?), but I was hoping there's some way to add this check in DataLoader_main.h.

Reply:
if you define a variable in `Setup` section of `training.yaml` it should be accessible in c++ (like, for example `n_tau` etc.)

Reply:
Yes, but my question was, what if it's not defined? It would cause a crash for everyone working with their current local configurations if they don't add that variable.

Reply:
oh, they will need to add it, otherwise, the code will crash. However, the change is minimal, so I think it is ok to apply it now (there is not that many people who actively run training and the patch for their configs will be trivial).

Reply:
I also think that this isn't a problem to add `debug` option. In general, each training is bound to use the version of DataLoader with which it was run. That means that every training is always in conflict with all future developments of DataLoader class. Basically, the main conflict point for current developments is `apply_training` when the DataLoader is used to load the data into the model. So to handle divergence of developments, there is a checkout of git commit used to run that given training:
   https://github.com/cms-tau-pog/TauMLTools/blob/d233f566dfd733fa01cdf7986feb816882555b09/Evaluation/apply_training.py#L51

In that perspective, your change won't affect the inference step of existing trainings, because in any case they use the past version of DataLoader (or in my case even more complicated workarounds, so your changes won't complicate them further at all :) )